lib/gen_lsp/test.ex

```diff
@@ -30,12 +30,28 @@ defmodule GenLSP.Test do
   """
   @spec server(mod :: atom()) :: server()
   def server(mod, opts \\ []) do
-    buffer =
-      start_supervised!({GenLSP.Buffer, communication: {GenLSP.Communication.TCP, [port: 0]}})
+    buffer_id = String.to_atom("#{opts[:name]}buffer")
+    lsp_id = String.to_atom("#{opts[:name]}lsp")
+
+    buffer = start_supervised!(
+      Supervisor.child_spec(
+        {
+          GenLSP.Buffer,
+          communication: {GenLSP.Communication.TCP, [port: 0]},
+          name: buffer_id
+        },
+        id: buffer_id
+      )
+    )
 
     {:ok, port} = :inet.port(GenLSP.Buffer.comm_state(buffer).lsocket)
 
-    lsp = start_supervised!({mod, Keyword.merge([buffer: buffer], opts)})
+    lsp = start_supervised!(
+      Supervisor.child_spec(
+        {mod, Keyword.merge([buffer: buffer], opts)},
+        id: lsp_id
+      )
+    )
```

Comment:
i don't think any of this needs to change

Reply:
Actually not as without specifying `id` I get the following failure:
```elixir
  1) test can start multiple servers in one test/2 (GenLSPTest)
     test/gen_lsp_test.exs:150
     ** (RuntimeError) failed to start child with the spec {GenLSP.Buffer, [communication: {GenLSP.Communication.TCP, [port: 0]}, name: :example_server_2buffer]}.
     Reason: bad child specification, more than one child specification has the id: GenLSP.Buffer.
     If using maps as child specifications, make sure the :id keys are unique.
     If using a module or {module, arg} as child, use Supervisor.child_spec/2 to change the :id, for example:

         children = [
           Supervisor.child_spec({MyWorker, arg}, id: :my_worker_1),
           Supervisor.child_spec({MyWorker, arg}, id: :my_worker_2)
         ]
```

Reply:
Or for the lsp without `id`:

```elixir
  1) test can start multiple servers in one test/2 (GenLSPTest)
     test/gen_lsp_test.exs:150
     ** (RuntimeError) failed to start child with the spec {GenLSPTest.ExampleServer, [buffer: #PID<0.288.0>, name: :example_server_2]}.
     Reason: bad child specification, more than one child specification has the id: GenLSPTest.ExampleServer.
     If using maps as child specifications, make sure the :id keys are unique.
     If using a module or {module, arg} as child, use Supervisor.child_spec/2 to change the :id, for example:

         children = [
           Supervisor.child_spec({MyWorker, arg}, id: :my_worker_1),
           Supervisor.child_spec({MyWorker, arg}, id: :my_worker_2)
         ]
```

Reply:
you can see in the NextLS PR that I had to specify this `id` in most of processes to make it work

Reply:
I think this is only happening because you are starting them under the same test supervisor, i don't think you'll ever be starting two LSPs in one test.

Reply:
I actually did it in the new test for NextLS 🙉